framework/src/CMakeLists.txt

```diff
@@ -90,4 +106,40 @@ set(_private_headers
   bundle/Debug.h
   bundle/Fragment.h
   bundle/Resolver.h
+
+  ../../third_party/absl/base/attributes.h
+  ../../third_party/absl/base/config.h
+  ../../third_party/absl/base/log_severity.h
+  ../../third_party/absl/base/macros.h
+  ../../third_party/absl/base/optimization.h
+  ../../third_party/absl/base/policy_checks.h
+  ../../third_party/absl/base/port.h
+  ../../third_party/absl/base/internal/atomic_hook.h
+  ../../third_party/absl/base/internal/raw_logging.h
+  ../../third_party/absl/base/internal/throw_delegate.h
+
+  ../../third_party/absl/strings/ascii.h
+  ../../third_party/absl/strings/numbers.h
+  ../../third_party/absl/strings/string_view.h
+  ../../third_party/absl/strings/internal/memutil.h
 )
+
+# Turn off warnings when compiling Abseil files
+file(GLOB_RECURSE ABSL_FILES 
+  "${CMAKE_SOURCE_DIR}/third_party/absl/*.cc" 
+  "${CMAKE_SOURCE_DIR}/third_party/absl/*.c" 
+  "${CMAKE_SOURCE_DIR}/third_party/absl/*.h")
+
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR 
+    "${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang" 
+    OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+  set(CF "-w")
+  set_source_files_properties(
```

Comment:
I'd suggest removing the existing -Werror and replacing it with -w only for the abseil source files.